IPs/RSKIP194.md

```diff
@@ -0,0 +1,108 @@
+
+Another motivation for this RSKIP is that the RSK block lacks a simple versioning system. Every new change requires using the number of fields to detect which type of block it is.
+As the number of modifications to the block header performed in hard forks increases, the changes can be mixed in different ways to create a high and unnecessary number of different types of blocks.
+This increases the complexity of validation code and makes unit testing really hard to cover all cases.
+
+This RSKIP adds to the block header a `blockVersion` field that allows clients to take decisions based on the blockVersion instead of trying to infer it.
+To detect this new field, the size of the bloom filter field is used, but hopefully after this change no other attributes of the fields will be used to detect the block version.
+
+
+## Specification
+
+The RSK block header is a list of the following elements:
+
+```
+parentHash, unclesHash, coinbase,stateRoot, txTrieRoot, receiptTrieRoot, logsBloom, difficulty, number, gasLimit, gasUsed, timestamp, extraData, paidFees, minimumGasPrice, uncleCount, ummRoot [, mergeMiningFields ] 
+```
+
+The *mergeMiningFields* are the following:
+
+```
+bitcoinMergedMiningHeader,
+bitcoinMergedMiningMerkleProof,
+bitcoinMergedMiningCoinbaseTransaction
+```
+
+All blocks prior the hard-fork block number are considered version 0.
+Afterward, new blocks must be version 1.
+For block headers version 1, there are two separate serialization methods, one for the transfer of the block header information and the other for the computation of the block hash.
+When serializing the block header version 1 to obtain the block hash, the field logsBloom is replaced by an RLP list of:
+
+* The blockVersion field (integer) (1 byte)
+* The Keccak hash of the bloom filter data (32 bytes).
+
+When serializing the block header for transmission or storage, the logsBloom field is: 
+* The blockVersion field (integer) (1 byte)
+* The bloom filter data (256 bytes).
+
+Consensus must validate that the blockVersion is exactly 1 for blocks after the hard fork, until this number is incremented by another consensus change.
+
+
+## Rationale
+
+Generally the blockVersion is stored as the first field of an object.
+While this is entirely possible for RSK block headers, PowHSMs cannot understand any other block header format. However, PowHSM can verify without problem a block header with shorter data stored in the logsBloom field.
+To avoid forcing a costly firmware upgrade for all PowHSMs, the blockVersion is stored in the logsBloom filter, which is unchecked by the hardware.
+
+While computing the hash without the full bloom filter data may incentivize nodes to remove this data permanently, the overhead of the bloom filter data is very low for blocks with many transaction. Therefore this incentive is weak. 
```

Comment:
Minor typo: "for blocks with many transaction" should be "for blocks with many transactions" (note the S at the end of transaction).